src/azul/portal_service.py

```diff
@@ -196,8 +200,11 @@ def _read_db(self, version: str) -> JSONs:
             response = self.client.get_object(Bucket=self.bucket,
                                               Key=self.object_key,
                                               VersionId=version)
-        except self.client.exceptions.NoSuchKey:
-            raise NoSuchObjectVersion(version)
+        except ClientError as e:
+            if 'Invalid version id specified' in e.response['Error']['Message']:
```

Comment:
Why was the changed? The use of the dynamically create exception classes superior to poking around in the response.

Reply:
@hannes-ucsc  

I need to fix this; there was a problem with the reproduction that I posted here. https://github.com/databiosphere/azul/issues/1718#issuecomment-801527437

There are two problems here, one was my replication of the invalid version key; i was using an invalid key, but it was also syntactically invalid. Using something like `invalid_version_key` will return a `Invalid version id specified response` not a `NoSuchObjectVersion` exception.

In order to get a NoSuchObjectVersion exception we have to pass a invalid version ID that is syntactically correct. (Something that our tests currently do NOT do https://github.com/DataBiosphere/azul/blob/28e48bb45eae0abaa4d9c4e9d72244cd4ea6e5d2/test/service/test_portal_service.py#L146)

It looks like moto 2.0.2 does NOT inspect for syntactically incorrect version IDs, (they use uuid4s for the version id, rather than the AWS version id)
https://github.com/spulec/moto/blob/f5e3cd891f59a4aa2db3b04fd14cb3e243a1ca52/moto/s3/responses.py#L1210

I need to double check to confirm that this is a bug with moto and raise an issue if my finding match my speculations.

Reply:
Replicated here: https://gist.github.com/amarjandu/334ffd668dd849991ba9949a32dcc3b8

Reply:
Raised: https://github.com/spulec/moto/issues/3884

Reply:
@hannes-ucsc the maintainers team decided that would prefer to keep the moto as feature incomplete (perform no validation) rather than inconsistent (s3api does not validate uuids as versionIDs). https://github.com/spulec/moto/pull/3887 was merged in as a fix to the incorrect exception that was being raised. It should be available as of moto >= 2.0.6.dev14 .

Should we hold off until 2.0.7 is available?